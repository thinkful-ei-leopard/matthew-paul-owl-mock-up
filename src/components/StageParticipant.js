import React from "react";

function StageParticipant(props) {
  return (
    <div className="Stage-Participant">
      <div className="Participant-Info-Container">
        <span className="Participant-Name">{props.name}</span>
      </div>
      <img className="Stage-Avatar" alt="avatar" src={props.avatar} />
    </div>
  );
}

export default StageParticipant;
