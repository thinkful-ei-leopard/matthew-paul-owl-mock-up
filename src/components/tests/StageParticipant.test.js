import React from 'react';
import ReactDOM from 'react-dom';
import StageParticipant from '../StageParticipant';
import renderer from 'react-test-renderer'; 

it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<StageParticipant key='test'
  name='test'
  avatar='https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1'
  />, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
  .create(<StageParticipant key='test'
      name='test'
      avatar='test'
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });