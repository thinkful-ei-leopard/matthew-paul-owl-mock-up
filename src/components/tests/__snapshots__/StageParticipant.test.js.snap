// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="stage-participant"
>
  <div
    className="participant-info-container"
  >
    <span
      className="participant-name"
    >
      test
    </span>
  </div>
  <img
    alt="avatar"
    className="stage-avatar"
    src="test"
  />
</div>
`;
